From 8aa6aa4ba2d9d00c91e36e146895af20cbc00d17 Mon Sep 17 00:00:00 2001
From: Brendan Heading <brendanheading@gmail.com>
Date: Wed, 2 Sep 2015 12:10:58 +0100
Subject: [PATCH 1/1] fix compilation issues with musl.

- sys/time.h needed for struct timeval
- unistd.h should not be included due to clash with encrypt(3)
- sys/types.h needed for mode_t

Note that this version of bluez is 3 years old. Recent fixes have gone
into the upstream to address musl compatibility issues, but the upstream
has changed so much that it makes little sense to try to backport these.
This patch is really a stopgap until we move to the latest upstream.

Upstream-Status: not needed
Signed-off-by: Brendan Heading <brendanheading@gmail.com>
---
 compat/bnep.c  | 1 +
 compat/dund.c  | 1 -
 src/textfile.h | 1 +
 3 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/compat/bnep.c b/compat/bnep.c
index 281350b..49226c1 100644
--- a/compat/bnep.c
+++ b/compat/bnep.c
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+#include <sys/time.h>
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/bnep.h>
diff --git a/compat/dund.c b/compat/dund.c
index af1b536..c2f085f 100644
--- a/compat/dund.c
+++ b/compat/dund.c
@@ -29,7 +29,6 @@
 #include <stdio.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
diff --git a/src/textfile.h b/src/textfile.h
index dc5fc2b..0148b30 100644
--- a/src/textfile.h
+++ b/src/textfile.h
@@ -24,6 +24,7 @@
 #ifndef __TEXTFILE_H
 #define __TEXTFILE_H
 
+#include <sys/types.h>
 int create_dirs(const char *filename, const mode_t mode);
 int create_file(const char *filename, const mode_t mode);
 int create_name(char *buf, size_t size, const char *path,
-- 
2.4.3

